src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -34,14 +54,60 @@ def __init__(self, msConfig, logger=None):
         """
         super(MSRuleCleaner, self).__init__(msConfig, logger=logger)
 
-        self.msConfig.setdefault("limitRequestsPerCycle", 500)
         self.msConfig.setdefault("verbose", True)
         self.msConfig.setdefault("interval", 60)
         self.msConfig.setdefault("services", ['ruleCleaner'])
-        self.msConfig.setdefault("rucioAccount", "FIXME_RUCIO_ACCT")
-        self.uConfig = {}
+        self.msConfig.setdefault("rucioWmaAccount", "wma_test")
+        self.msConfig.setdefault("rucioMStrAccount", "wmcore_transferor")
+        self.msConfig.setdefault('enableRealMode', False)
+        self.mode = "RealMode" if self.msConfig['enableRealMode'] else "DryRunMode"
         self.emailAlert = EmailAlert(self.msConfig)
 
+        # Building all the Pipelines:
+        pName = 'plineMSTrCont'
+        self.plineMSTrCont = Pipeline(name=pName,
+                                      funcLine=[Functor(self.setPlineMarker, pName),
+                                                Functor(self.cleanRucioRules)])
+        pName = 'plineMSTrBlock'
+        self.plineMSTrBlock = Pipeline(name=pName,
+                                       funcLine=[Functor(self.setPlineMarker, pName),
+                                                 Functor(self.cleanRucioRules)])
+        pName = 'plineAgentCont'
+        self.plineAgentCont = Pipeline(name=pName,
+                                       funcLine=[Functor(self.setPlineMarker, pName),
+                                                 Functor(self.getMSOutputTransferInfo),
+                                                 Functor(self.getRucioRules, 'container', self.msConfig['rucioWmaAccount']),
+                                                 Functor(self.cleanRucioRules)])
+        pName = 'plineAgentBlock'
+        self.plineAgentBlock = Pipeline(name=pName,
+                                        funcLine=[Functor(self.setPlineMarker, pName),
+                                                  Functor(self.getMSOutputTransferInfo),
+                                                  Functor(self.getRucioRules, 'block', self.msConfig['rucioWmaAccount']),
+                                                  Functor(self.cleanRucioRules)])
+        pName = 'plineArchive'
+        self.plineArchive = Pipeline(name=pName,
+                                     funcLine=[Functor(self.setPlineMarker, pName),
+                                               Functor(self.setClean),
+                                               Functor(self.archive)])
+
+        # Building the different set of plines we will need later:
+        # NOTE: The following are all the functional pipelines which are supposed to include
+        #       a cleanup function and report cleanup status in the MSRuleCleanerWflow object
+        self.cleanuplines = [self.plineMSTrCont,
```

Comment:
I am not quite sure how exactly that  would work. Because am having those as object bound variables not because I access them from different pipelines from the same instance, but rather to use them as predefined lists/sest of different pipelines, which are not accessed from the `__init__()` but from a method down in the code. And I on the other hand did not want to recreate those lists every time a workflow enters the `_dispatcher()` ( which would be the only way to accomplish what you are suggesting here - IIUC), but rather keep those in the instance and just refer to the proper set/list of pipelines based on the criteria applied on the workflow inside the `_dispatcher()`. Or if I got you wrong, Please correct me.